bitcoin/examples/ecdsa-psbt-simple.rs

```diff
@@ -0,0 +1,246 @@
+    let pk_inputs = [pk_input_1, pk_input_2];
+    let pk_change = Xpub::from_priv(&secp, &xpriv_change).to_pub();
+
+    // Get the Witness Public Key Hashes (WPKHs)
+    let wpkhs: Vec<WPubkeyHash> = pk_inputs.iter().map(|pk| pk.wpubkey_hash()).collect();
+
+    // Get the unspent outputs that is locked to the key above that we control.
+    // In a real application these would come from the chain.
+    let utxos: Vec<TxOut> =
+        dummy_unspent_transaction_outputs().into_iter().map(|(_, utxo)| utxo).collect();
+
+    // Get the addresses to send to.
+    let address = receivers_address();
+
+    // The inputs for the transaction we are constructing.
+    let inputs: Vec<TxIn> = dummy_unspent_transaction_outputs()
+        .into_iter()
+        .map(|(outpoint, _)| TxIn {
+            previous_output: outpoint,
+            script_sig: ScriptBuf::default(),
+            sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+            witness: Witness::default(),
+        })
+        .collect();
+
+    // The spend output is locked to a key controlled by the receiver.
+    let spend = TxOut { value: SPEND_AMOUNT, script_pubkey: address.script_pubkey() };
+
+    // The change output is locked to a key controlled by us.
+    let change = TxOut {
+        value: CHANGE_AMOUNT,
+        script_pubkey: ScriptBuf::new_p2wpkh(&pk_change.wpubkey_hash()), // Change comes back to us.
+    };
+
+    // The transaction we want to sign and broadcast.
+    let unsigned_tx = Transaction {
+        version: transaction::Version::TWO,  // Post BIP 68.
+        lock_time: absolute::LockTime::ZERO, // Ignore the locktime.
+        input: inputs,                       // Input is 0-indexed.
+        output: vec![spend, change],         // Outputs, order does not matter.
+    };
+
+    // Now we'll start the PSBT workflow.
+    // Step 1: Creator role; that creates,
+    // and add inputs and outputs to the PSBT.
+    let mut psbt = Psbt::from_unsigned_tx(unsigned_tx.clone()).expect("Could not create PSBT");
+
+    // Step 2:Updater role; that adds additional
+    // information to the PSBT.
+    let ty = PsbtSighashType::from_str("SIGHASH_ALL").expect("valid sighash type");
+    let derivation_paths = [
+        DerivationPath::from_str("m/84'/0'/0'/0/0").expect("valid derivation path"),
+        DerivationPath::from_str("m/84'/0'/0'/1/0").expect("valid derivation path"),
+    ];
+    let mut bip32_derivations = Vec::new();
+    for (idx, pk) in pk_inputs.iter().enumerate() {
+        let mut map = BTreeMap::new();
+        let fingerprint = Fingerprint::from_str(MASTER_FINGERPRINT).expect("valid fingerprint");
+        map.insert(pk.0, (fingerprint, derivation_paths[idx].clone()));
+        bip32_derivations.push(map);
+    }
+    psbt.inputs = vec![
+        Input {
+            witness_utxo: Some(utxos[0].clone()),
+            redeem_script: Some(ScriptBuf::new_p2wpkh(&wpkhs[0])),
+            bip32_derivation: bip32_derivations[0].clone(),
```

Comment:
In 6564ca14397678a8c0a334bef5a6049013a634c6

I'm sure you could avoid allocating a vector of maps and then cloning items to destructure it, but for example code this might be the clearest thing.

Reply:
ha, yeah I pondered the loop to set up `bip32_derivations`. I think this code is as clear as it gets. My main concern is there is an implicit dependency between the `DerivationPath` strings (and order) and code way up at the top of the file. I could not come up with anything cleaner though except putting the fingerprint construction outside the loop (to improve clarity and efficiency).